app/migrations/Version20180531173942.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+final class Version20180531173942 extends AbstractMigration
+{
+    public function up(Schema $schema): void
+    {
+        $this->addSql('CREATE TABLE mooc_chapter (id INT UNSIGNED AUTO_INCREMENT NOT NULL, mooc_id INT UNSIGNED DEFAULT NULL, name VARCHAR(255) NOT NULL, slug VARCHAR(255) NOT NULL, published TINYINT(1) NOT NULL, published_at DATETIME NOT NULL, display_order SMALLINT DEFAULT 1 NOT NULL, INDEX IDX_A3EDA0D1255EEB87 (mooc_id), UNIQUE INDEX mooc_chapter_slug (slug), UNIQUE INDEX mooc_chapter_order_display_by_mooc (display_order, mooc_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8 COLLATE UTF8_unicode_ci ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE mooc_elements (id INT UNSIGNED AUTO_INCREMENT NOT NULL, chapter_id INT UNSIGNED DEFAULT NULL, name VARCHAR(255) NOT NULL, slug VARCHAR(255) NOT NULL, display_order SMALLINT DEFAULT 1 NOT NULL, created_at DATETIME NOT NULL, updated_at DATETIME NOT NULL, type VARCHAR(255) NOT NULL, youtube_url VARCHAR(255) DEFAULT NULL, youtube_thumbnail_url VARCHAR(255) DEFAULT NULL, INDEX IDX_691284C5579F4768 (chapter_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8 COLLATE UTF8_unicode_ci ENGINE = InnoDB');
```

Comment:
no it's nullable, because it's possible to "detach" a video from a chapter and set null the `chapter_id` field.

https://github.com/EnMarche/en-marche.fr/pull/2475/files#diff-9c03ca5375c18fa618fe02cb79c8a7aaR192

Same for the Mooc and Chapter.